File: U404969.py
from math import *
a = input()
a = a.split()
x = int(a[0])
n = int(a[1])
ans = 0
if x == 2 and n > 1:
    print("inf") 
    exit(0)
else:
    if x >= pow(2,n):
        print(ans)
    else:
        n -= 2
        ans = n / (log(x,2) - 1)
    
        ansWer =ceil(log(ans,2))

        print(ansWer)

File: U404969.cpp
#include <bits/stdc++.h>
using namespace std;
double x, n;
double ans;
int ansWer;
int main()
{
    cin >> x >> n;
    if(x == 2 && n > 2) {cout << "inf"; return 0;}
    if(x == 2 && n == 1) {cout << 0; return 0;}
    if(x == 2 && n == 2) {cout << 1; return 0;}
    n -= 2 ;

    ans = n / (log2(x) - 1);
    
    ansWer =ceil(log2(ans));
    cout << ansWer;
    system("pause");
    return 0;
}